scripts/cicd-canary/update_secret_for_networkpolicy_canary.sh

```diff
@@ -112,27 +112,108 @@ function getAppEnvironments() {
 
 function updateSecret() {
     app_env=$1
-    printf "Updating NETWORKPOLICY_CANARY_PASSWORD for environment $app_env \n"
+    secret_name=$2
+    secret_value=$3
+    radix_component=$4
+    printf "Updating ${secret_name} for environment ${app_env} in ${radix_component} component \n"
     API_REQUEST=$(curl \
          --silent \
          -X PUT \
-         "https://server-radix-api-prod.${CLUSTER_NAME}.${AZ_RESOURCE_DNS}/api/v1/applications/radix-networkpolicy-canary/environments/${app_env}/components/web/secrets/NETWORKPOLICY_CANARY_PASSWORD" \
+         "https://server-radix-api-prod.${CLUSTER_NAME}.${AZ_RESOURCE_DNS}/api/v1/applications/radix-networkpolicy-canary/environments/${app_env}/components/${radix_component}/secrets/${secret_name}" \
          -H "accept: application/json" \
          -H "Authorization: bearer ${API_ACCESS_TOKEN}" \
          -H "Content-Type: application/json" \
-         -d "{ \"secretValue\": \"${NETWORKPOLICY_CANARY_PASSWORD}\"}")
+         -d "{ \"secretValue\": \"${secret_value}\"}")
     if [[ "${API_REQUEST}" != "\"Success\"" ]]; then
         echo -e "\nERROR: API request failed."  >&2
         exit 1
     fi
     printf "Secret updated\n"
 }
 
+function resetAppRegistrationPassword() {
+    # Generate new secret for App Registration.
+    printf "Re-generate client secret for App Registration \"$APP_REGISTRATION_NETWORKPOLICY_CANARY\"...\n"
+    APP_REGISTRATION_CLIENT_ID=$(az ad app list --display-name "$APP_REGISTRATION_NETWORKPOLICY_CANARY" | jq -r '.[].appId')
+
+    UPDATED_APP_REGISTRATION_PASSWORD=$(az ad app credential reset --id "$APP_REGISTRATION_CLIENT_ID" --credential-description "${RADIX_ZONE}-${RADIX_ENVIRONMENT}" --append 2>/dev/null | jq -r '.password') # For some reason, description can not be too long.
+    if [[ -z "$UPDATED_APP_REGISTRATION_PASSWORD" ]]; then
+        echo -e "\nERROR: Could not re-generate client secret for App Registration \"$APP_REGISTRATION_NETWORKPOLICY_CANARY\". Exiting..." >&2
+        exit 1
+    fi
```

Comment:
Output error from command and return 1 on error.
```suggestion
    # For some reason, description can not be too long.
    UPDATED_APP_REGISTRATION_PASSWORD=$(az ad app credential reset \
        --id "$APP_REGISTRATION_CLIENT_ID" \
        --credential-description "${RADIX_ZONE}-${RADIX_ENVIRONMENT}" \
        --append \
        --query password \
        --output tsv \
        --only-show-errors) || { echo -e "\nERROR: Could not re-generate client secret for App Registration \"$APP_REGISTRATION_NETWORKPOLICY_CANARY\"." >&2; return 1; }
```